parsec/scheduling.c

```diff
@@ -532,6 +531,108 @@ int __parsec_task_progress( parsec_execution_stream_t* es,
     return rc;
 }
 
+static int __parsec_taskpool_test( parsec_taskpool_t* tp, parsec_execution_stream_t *es )
+{
+    parsec_context_t* parsec_context = es->virtual_process->parsec_context;
+    parsec_task_t* task;
+    int nbiterations = 0, distance, rc;
+
+    assert(PARSEC_THREAD_IS_MASTER(es));
+
+    /* first select begin, right before the wait_for_the... goto label */
+    PARSEC_PINS(es, SELECT_BEGIN, NULL);
+
+    if( NULL == parsec_current_scheduler ) {
+        parsec_fatal("Main thread entered parsec_taskpool_test, while a scheduler is not selected yet!");
+        return -1;
+    }
+
+    if( tp->tdm.module->taskpool_state(tp) != PARSEC_TERM_TP_TERMINATED ) {
+        /* Here we detach all dtd taskpools registered with us */
+        parsec_detach_all_dtd_taskpool_from_context(parsec_context);
```

Comment:
Shouldn't this do the same as `test` below?
```
/* Detach the current taskpool from the associated context */
     parsec_dtd_dequeue_taskpool(tp);
     PARSEC_LIST_ITEM_SINGLETON(tp);
```

Reply:
With the new API we designed few days ago, the use of `parsec_detach_all_dtd_taskpool_from_context` will not be necessary.